### ISSUE TITLE
Validation: Search rules just for first path that fits, or for first path that fits and specifies key?

### ISSUE THREAD

**Issue by TobiasKadelka** [psyinfra/onyo#106]
When having a `validation.yaml` file specifying the following rules:

```
"shelf/**":
- USB:
  - Type: int
"*/**":
- RAM:
  - Type: int
```

And a file `shelf/laptop_apply_macbook.1` that looks like:
```
---
RAM: 12GB
```

- Should the second rule, that specifies RAM as `int` say that the asset is invalid, because it is the first rule saying something about the RAM value, where the name scheme fits?
- Or should it except it as valid, because "shelf/**" is the first name scheme that fits, and it does not say anything about that key?

The question is, should just the first name scheme that fits the file be applied (and basically ignoring later rules, because `shelf/**` did say nothing specific about that key), or should a later rule be applied if there are no earlier rules that fit the specific key?

The difference is having `*/**` as a general fallback for all files in onyo, or to just use it, when a file has no rules otherwise defined.

**Comment by TobiasKadelka** [comment]
Update: for the PR #100 I decided, that all assets that match shelf/** follow just the rules defined in shelf.
If the rules under shelf/** say nothing about a key RAM, then the rule does not apply, no matter what later rules under */** say.

This depends just on a single line in the code, so if this turns out to be inconvenient, I can easily change it (the effort to change the README.md is the biggest step).

**Comment by TobiasKadelka** [comment]
We should really discuss this again, since the implementation discribed makes the rules for `shelf/**` irrelevant, if already a case `shelf/*` is defined:
```
"shelf/*":                                                                       
- RAM:                                                                           
  - Type: int                                                                  
"shelf/**":                                                                      
- Size:                                                                          
  - Type: int
```

This is not necessarily bad, but it should be talked about in more detail, when rules should apply and when not.

**Comment by TobiasKadelka** [comment]
It was discussed, that we will apply just the first path that matches, because it will make it easier for the user to find the rule that applies to any given file.
This is implemented with #100
